File: frontend/src/utils/etablissementsConnus.ts
/** Catalogue d'établissements connus (refonte import, 05/09/2026, demande directe
 * de l'utilisateur : « une liste des établissements... et en ajouter des
 * customs »), consommé par `CatalogueEtablissementPicker`/`EtablissementLogo`.
 *
 * Chaque entrée est un badge GÉNÉRÉ (initiales + couleur de marque approximative),
 * jamais un logo réel : reproduire les logos officiels poserait un risque de
 * droits et demanderait de télécharger des images depuis internet — un badge
 * généré suffit à distinguer visuellement les établissements sans ce risque.
 * `cle` est stockée telle quelle dans `Etablissement.logo_key` ; une clé absente de
 * ce catalogue (établissement personnalisé, ou catalogue qui aurait changé) retombe
 * simplement sur le badge neutre de `EtablissementLogo`. */
export interface EtablissementConnu {
  cle: string
  nom: string
  couleur: string
  initiales: string
}

export const CATALOGUE_ETABLISSEMENTS: EtablissementConnu[] = [
  { cle: 'trade_republic', nom: 'Trade Republic', couleur: '#1b1b1f', initiales: 'TR' },
  { cle: 'boursorama', nom: 'Boursorama Banque', couleur: '#e2001a', initiales: 'BO' },
  { cle: 'bourse_direct', nom: 'Bourse Direct', couleur: '#0057a3', initiales: 'BD' },
  { cle: 'degiro', nom: 'Degiro', couleur: '#003d4c', initiales: 'DG' },
  { cle: 'interactive_brokers', nom: 'Interactive Brokers', couleur: '#d4310a', initiales: 'IB' },
  { cle: 'saxo', nom: 'Saxo Banque', couleur: '#2d2d87', initiales: 'SX' },
  { cle: 'fortuneo', nom: 'Fortuneo', couleur: '#6a2c8f', initiales: 'FO' },
  { cle: 'bforbank', nom: 'BforBank', couleur: '#e01e5a', initiales: 'BF' },
  { cle: 'credit_agricole', nom: 'Crédit Agricole', couleur: '#00895e', initiales: 'CA' },
  { cle: 'societe_generale', nom: 'Société Générale', couleur: '#e30613', initiales: 'SG' },
  { cle: 'bnp_paribas', nom: 'BNP Paribas', couleur: '#00915a', initiales: 'BN' },
  { cle: 'caisse_epargne', nom: "Caisse d'Épargne", couleur: '#e2001a', initiales: 'CE' },
]

export function trouverEtablissementConnu(cle: string | null | undefined): EtablissementConnu | undefined {
  if (!cle) return undefined
  return CATALOGUE_ETABLISSEMENTS.find((e) => e.cle === cle)
}

File: backend/app/services/etablissements_connus.py
"""Domaine officiel de chaque établissement du catalogue (retour utilisateur du
05/09/2026 : « avoir les logos à jour »), utilisé par `logo_service` pour aller
chercher le logo réel sur le site de l'établissement lui-même.

Pourquoi le site officiel plutôt qu'une banque d'icônes tierce : la piste
initialement proposée (dashboardicons.com) a été vérifiée avant d'être écartée —
sur les 14 établissements envisagés, seuls Revolut et N26 y existent, et le set
entier ne contient aucune entrée « bank/banque/bourse/broker/agricole/paribas… ».
C'est un catalogue pensé pour les applications auto-hébergées, pas pour les
banques européennes. Le favicon/apple-touch-icon du site officiel, lui, existe
pour tous, ne dépend d'aucun tiers et ne signale à personne quelles banques le
foyer utilise.

Les CLÉS doivent rester identiques à celles de `frontend/src/utils/etablissementsConnus.ts`
(qui porte, lui, le nom affiché, la couleur et les initiales du badge de repli).
Une clé présente côté frontend mais absente ici ne casse rien : elle n'a
simplement pas de récupération automatique, et garde son badge généré.
"""

# Domaines vérifiés un par un en conditions réelles le 05/09/2026 : 11 des 12
# rendent bien une icône exploitable. Seul BNP Paribas répond 403 à tout (protection
# anti-robot, y compris sur `/favicon.ico`) — sans conséquence, il garde simplement
# son badge généré tant que l'utilisateur ne lui fournit pas une image lui-même.
# `societe_generale` pointe vers le sous-domaine particuliers : le domaine racine ne
# répond pas, et `www.` non plus.
DOMAINES: dict[str, str] = {
    "trade_republic": "traderepublic.com",
    "boursorama": "boursobank.com",
    "bourse_direct": "boursedirect.fr",
    "degiro": "degiro.fr",
    "interactive_brokers": "interactivebrokers.com",
    "saxo": "home.saxo",
    "fortuneo": "fortuneo.fr",
    "bforbank": "bforbank.com",
    "credit_agricole": "credit-agricole.fr",
    "societe_generale": "particuliers.societegenerale.fr",
    "bnp_paribas": "mabanque.bnpparibas",
    "caisse_epargne": "caisse-epargne.fr",
}


def domaine_pour(logo_key: str | None) -> str | None:
    if not logo_key:
        return None
    return DOMAINES.get(logo_key)
